lib/pure/json.nim

```diff
@@ -1235,6 +1229,8 @@ proc processType(typeName: NimNode, obj: NimNode,
       result = quote do:
         verifyJsonKind(`jsonNode`, {JObject, JNull}, astToStr(`jsonNode`))
         if `jsonNode`.kind == JNull:
+          # JNull should _not_ be implicitly converted to a nil ref?
+          # per https://github.com/nim-lang/Nim/issues/11894
```

Comment:
I left this `nil` -> `nil ref` in place because without it, the breakage might be too extreme.  But, this would be one of those places where you might want to cut deeper.

Reply:
This will conflict with my pull request:
https://github.com/nim-lang/Nim/pull/12391/files#diff-e00d77321d9bf30eaf8e4be863d9b565L1189